packages/shared/src/uops/protocols/eip-7821/methods.ts

```diff
@@ -69,3 +80,43 @@ function executionModeId(mode: string): number {
       throw new Error(`Unsupported execution mode: ${mode}`)
   }
 }
+
+function decodeBatchInput(data: `0x${string}`): TransferOperation[] {
+  if (!data) {
+    return []
+  }
+
+  const reader = new BinaryReader(data)
+  reader.read(4)
```

Comment:
maybe offset instead reading and voiding results?

```suggestion
  const reader = new BinaryReader(data, {offset: 4})
```

Reply:
Done